event_response_tool/event_response.py

```diff
@@ -0,0 +1,323 @@
+                analysis_profile_description=self.event_response_inputs.analysis_profile_description,
+                old_analysis_profile_uuid=self.event_response_inputs.old_analysis_profile_uuid,
+            )
+            if self.event_response_inputs.old_analysis_profile_uuid:
+                ap_creator.update_analysis_profile()
+            else:
+                ap_creator.create_analysis_profile()
+
+
+# Set up logging
+def init_logging():
+    logging.config.fileConfig("logging.ini")
+    logger = logging.getLogger(__name__)
+
+
+def init_directories():
+    # Create unique output directory for every attempt (FORMAT: archive/event_response-[datetime])
+    # to store the output files and log
+    now = datetime.now()
+    current_datetime_string = now.strftime("%d-%m-%Y-%H:%M:%S")
+    target_dir = "{}-{}".format("event_response", current_datetime_string)
+    output_directory = config.get("defaults", "output_directory")
+    full_output_dir = "{}/{}".format(output_directory, target_dir)
+
+    try:
+        os.makedirs(full_output_dir, exist_ok=True)
+    except OSError as e:
+        alert.exception(
+            f"Creation of {full_output_dir} directory failed: {e}"
+        )
+    else:
+        if os.path.isfile("activity.log"):
+            shutil.move("activity.log", f"{full_output_dir}/activity.log")
+        return full_output_dir
+
+
+def check_empty_credentials(url, user_name, password):
+    updated_url = url if url else config.get("server", "base_url")
+    updated_username = (
+        user_name if user_name else config.get("server", "username")
+    )
+    updated_password = (
+        password if password else config.get("server", "password")
+    )
+
+    return updated_url, updated_username, updated_password
+
+
+def login(url, username, password):
+    try:
+        analyzere.base_url = url
+        analyzere.username = username
+        analyzere.password = password
+
+        analyzere.EventCatalog.list(limit=0)
+    except Exception as e:
+        alert.exception(f"Unable to connect to {url}: {e}")
+    else:
+        alert.info(f"Log in successful for user {username}")
+
+
+def validate_inputs(event_response_inputs):
+    # Check for required input
+    if (
+        event_response_inputs.event_weights_csv is None
+        or len(event_response_inputs.event_weights_csv) == 0
+    ):
+        alert.error(
+            "Please input the path of the CSV containing event weights"
+        )
+
+    # If layer_view CSV or portfolio_view UUID is not provided,
+    # check if analysis profile inputs are available
+    if (
+        event_response_inputs.layer_views_csv is None
+        or len(event_response_inputs.layer_views_csv) == 0
+    ) and (
+        event_response_inputs.portfolio_view_uuid is None
+        or len(event_response_inputs.portfolio_view_uuid) == 0
+    ):
+        if (
+            (event_response_inputs.total_number_of_events is None)
+            or (event_response_inputs.trial_count is None)
```

Comment:
is only one of `total_number_of_events` and `trial_count` needed? If so I think that could be more clear in the notebook/CLI helper comments